docs/en/interfaces/formats/Protobuf/Protobuf.md

```diff
@@ -85,6 +85,43 @@ message MessageType {
 }
 ```
 
+
+If a message contains [oneof](https://protobuf.dev/programming-guides/proto3/#oneof) and `input_format_protobuf_oneof_presence` is set, ClickHouse fills column that indicates which field of oneof was found.
+
+```capnp
+syntax = "proto3";
+
+message StringOrString {
+  oneof string_oneof {
+    string string1 = 1;
+    string string2 = 42;
+  }
+}
+```
+
+```sql
+CREATE TABLE string_or_string ( string1 String, string2 String, string_oneof Enum('no'=0, 'hello' = 1, 'world' = 42))  Engine=MergeTree ORDER BY tuple();
+INSERT INTO string_or_string from INFILE '$CURDIR/data_protobuf/String1' SETTINGS format_schema='$SCHEMADIR/string_or_string.proto:StringOrString' FORMAT ProtobufSingle;
+SELECT * FROM string_or_string
+```
+
+```text
+   ┌─────────┬─────────┬──────────────┐
+   │ string1 │ string2 │ string_oneof │
+   ├─────────┼─────────┼──────────────┤
+1. │         │ string2 │ world        │
+   ├─────────┼─────────┼──────────────┤
+2. │ string1 │         │ hello        │
+   └─────────┴─────────┴──────────────┘
+
+```
+Name of the column that indicates presence must be the same as the name of oneof. Nested messages are supported (see  [basic_examples](#basic_examples).
```

Comment:
Right bracket is missed.
Seems too weak reason to rerun build, waiting for other findings/suggestions.